packages/dds/merge-tree/src/client.ts

```diff
@@ -842,6 +851,80 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 		return this._mergeTree.getPosition(segment, perspective);
 	}
 
+	/**
+	 * Rebases a sided local reference to the best fitting position in the current tree.
+	 */
+	private rebaseSidedLocalReference(
+		ref: LocalReferencePosition,
+		reconnectingPerspective: Perspective,
+		slidePreference: SlidingPreference,
+	): RebasedObliterateEndpoint {
+		const oldSegment = ref.getSegment();
+		const oldOffset = ref.getOffset();
+		assert(oldSegment !== undefined && oldOffset !== undefined, "Invalid old reference");
+		const useNewSlidingBehavior = true;
+		// Destructuring segment + offset is convenient and segment is reassigned
+		// eslint-disable-next-line prefer-const
+		let { segment: newSegment, offset: newOffset } = getSlideToSegoff(
+			{ segment: oldSegment, offset: oldOffset },
+			slidePreference,
+			reconnectingPerspective,
+			useNewSlidingBehavior,
+		);
+
+		newSegment ??=
+			slidePreference === SlidingPreference.FORWARD
+				? this._mergeTree.endOfTree
+				: this._mergeTree.startOfTree;
+
+		assert(
+			isSegmentLeaf(newSegment) && newOffset !== undefined,
+			"Invalid new segment on rebase",
+		);
+
+		const { side } = (ref.properties as { side: Side }) ?? {};
+		assert(side !== undefined, "Side should have been set on sided local reference");
+
+		const newSide: Side =
+			newSegment === oldSegment
+				? side
+				: // If the reference slid to a new position, the closest fit to the original position will be independent of
```

Comment:
is this true, we always pass useNewSlidingBehavior, so shouldn't we always slide in the specified direction, we might just land at an endpoint

Reply:
We could try to slide via the direction specified, but it means we may obliterate segments which wouldn't have been obliterated if the original submission went through. It makes subsequent rebase logic a bit harder but should be doable.

See the "obliterate shrinks end point to best possible option" test case for an example of what I'm talking about here.

Reply:
ok. that makes sense. i'm fine with it as is.